File: src/compiler/typescript_to_rust/generators/ic_object/functions/print.ts
import { Rust } from '../../../../../types';

export function generateIcObjectFunctionPrint(): Rust {
    return `
        fn azle_ic_print(
            _this: &boa_engine::JsValue,
            _aargs: &[boa_engine::JsValue],
            _context: &mut boa_engine::Context
        ) -> boa_engine::JsResult<boa_engine::JsValue> {
            ic_cdk::println!("{:#?}", _aargs);
        
            return Ok(boa_engine::JsValue::Undefined);
        }
    `;
}

File: src/compiler/typescript_to_rust/azle_generate/src/body/ic_object/functions/print.rs
pub fn generate() -> proc_macro2::TokenStream {
    quote::quote! {
        fn print(
            _this: &boa_engine::JsValue,
            aargs: &[boa_engine::JsValue],
            context: &mut boa_engine::Context,
        ) -> boa_engine::JsResult<boa_engine::JsValue> {
            let output_string = aargs
                .iter()
                .map(|val| val.clone().to_console_string(context))
                .collect::<Vec<String>>()
                .join(" ");

            ic_cdk::println!("{}", output_string);

            return Ok(boa_engine::JsValue::Undefined);
        }
    }
}
